includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -252,35 +255,65 @@ private function process_alternate_names( $attr_spec_list ) {
 	 * @since 0.5
 	 */
 	public function sanitize() {
+		$result = $this->sanitize_element( $this->root_element );
+		if ( is_array( $result ) ) {
+			$this->script_components = $result;
+		}
+	}
 
-		// Add root of content to the stack.
-		$this->stack[] = $this->root_element;
-
-		/**
-		 * This loop traverses through the DOM tree iteratively.
-		 */
-		while ( ! empty( $this->stack ) ) {
+	/**
+	 * Sanitize element.
+	 *
+	 * Walk the DOM tree with depth first search (DFS) with post order traversal (LRN).
+	 *
+	 * @param DOMElement $element Element.
+	 * @return string[]|null Required component scripts from sanitizing an element tree, or null if the element was removed.
+	 */
+	private function sanitize_element( DOMElement $element ) {
+		if ( ! isset( $this->open_elements[ $element->nodeName ] ) ) {
+			$this->open_elements[ $element->nodeName ] = 0;
+		}
+		$this->open_elements[ $element->nodeName ]++;
+
+		$script_components = [];
+
+		// First recurse into children to sanitize descendants.
+		// The check for $element->parentNode at each iteration is to make sure an invalid child didn't bubble up removal
```

Comment:
Something off with the phrase here.
Maybe like this:
```suggestion
		// The check for $element->parentNode at each iteration is to make sure an invalid child didn't bubble up removed
```